kie-api/src/main/java/org/kie/api/runtime/rule/QueryResultsRow.java

```diff
@@ -21,4 +21,18 @@
  */
 public interface QueryResultsRow extends Row {
 
+    /**
+     * Return the Object for the given Declaration identifier.
+     * @param identifier
+     * @return the Object
+     */
+    public Object get(String identifier);
+
+    /**
+     * Return the {@link FactHandle} for the given Declaration identifier.
+     * @param identifier
+     * @return the {@link FactHandle} instance
+     */
+    public FactHandle getFactHandle(String identifier);
+
```

Comment:
Question about backwards compatibility: I am assuming the interface `QueryResultsRow` should _not_ be directly implemented by end-users, they will only use concrete instances of classes implementing this interface. So adding new methods to the interface should not actually break backwards compatibility. Is that correct?